source/features/highest-rated-comment.tsx

```diff
@@ -4,48 +4,80 @@ import select from 'select-dom';
 import features from '../libs/features';
 import * as icons from '../libs/icons';
 
-function getCount(reaction: HTMLElement): number {
-	return Number(/\d+/.exec(reaction.textContent!)![0]);
-}
+// `.js-timeline-item` gets the nearest comment excluding the very first comment (OP post)
+const COMMENT_SELECTOR = '.js-timeline-item';
+
+const positiveReactions = [
+	'[aria-label*="reacted with thumbs up"]',
+	'[aria-label*="reacted with hooray"]',
+	'[aria-label*="reacted with heart"]'
+];
+
+const negativeReactions = [
+	'[aria-label*="reacted with thumbs down"]'
+];
 
 function init(): false | void {
+	const bestComment = getBestComment();
+	if (!bestComment) {
+		return false;
+	}
+
+	highlightBestComment(bestComment);
+	linkBestComment(bestComment);
+}
+
+function getBestComment(): HTMLElement | null {
 	let highest;
-	// `.js-timeline-item` excludes the very first comment
-	for (const like of select.all('.js-timeline-item [aria-label*="reacted with thumbs up"]')) {
-		const count = getCount(like);
-		const dislike = select('[aria-label*="reacted with thumbs down"]', like.parentElement!);
+	for (const posReaction of getWatchedReactions()) {
```

Comment:
I think here you're getting the same comment up to 3 times. I'd rather see:

```
for each '.js-timeline-item'
	count positive reactions
	if count < 10
		continue
	count negative reactions
	... ignore controversial, etc
endfor
```

Reply:
Yup that is intentional. I actually tried iterating over all the comments but I found there was a significant slow down compared to going over only the positive reactions.

Reply:
Then I'd still suggest this

```rb
for each commentsWithPositiveReactions
	count positive reactions
	if count < 10
		continue
	count negative reactions
	... ignore controversial, etc
end
```

`commentsWithPositiveReactions` is something like:

```js
new Set(getPositiveReactions().map(reaction => reaction.closest('comment')))
```

This Set will include each unique comment, not the reactions.

This way it will skip any counting if the comment doesn't pass the base requirement of 10 votes, which most comments don't.

Reply:
Sure, I'll test it out. I was initially worried because I thought that since Sets use reference equality then perhaps the results from two different calls might have different references and fail the equality check but maybe that's not the case.